### PR TITLE
Math magicians: Calculator component

### PR DESCRIPTION
**Pull request by jash-debug**
- [ ] Delete boilerplate files
- [ ] Create the Component file
- [ ] Create the calculator React component
- [ ] Add minimal styling
- [ ] Fix linter errors
- [ ] Update descriptive readme

### CODE REVIEW

**Review by AlphaNtihinduka — CHANGES_REQUESTED** [review]
## Hi @jash-debug 🙋‍♂️
Good job working on implementing the calculator components. You have done the design to do the expectation and incredibly well. 👏  

TO HIGHLIGHT:
- Your project is professional ✅
- Professional README file :100: 
- Your PR is not yet professional ✅
- Well-structured files ✅

# STATUS: Required Changes :recycle: 

close

## Required changes :recycle: 
- [ ] I have noticed that your naming convention may need improvement. I suggest to rename `calculator.js` to `Calculator.js`.The name of that component has to begin with a capital letter otherwise React may throw an error as unrecognized tag. This convention is because only HTML elements and SVG tags can begin with a lowercase letter.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or slack channel if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK :crossed_fingers: :four_leaf_clover:

File: src/components/calculator.js
Comment (Shisui6) [inline src/components/calculator.js:1]: As the previous reviewer suggested, please rename this file to `Calculator.js`. This is one of the project requirements, and without it, your project does not run as seen in the image below
image

File: src/components/calculator.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  render() {
```
Comment (Shisui6) [inline src/components/calculator.js]: [Optional] This is one of the project requirements but was missed by the previous reviewer so I highly encourage you to implement it.

- Please update your calculator component to be a functional component and not a class component.
image

File: src/components/calculator.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    return (
+      <div id="calculator">
+        <input type="text" value={0} className="screen" />
+        <div id="numbers">
+          <button type="button" id="AC" value="AC" onClick={null}>Ac</button>
+          <button type="button" id="+/-" value="+/-" onClick={null}>+/-</button>
+          <button type="button" id="%" value="%" onClick={null}>%</button>
+          <button type="button" id="+" value="+" className="special" onClick={null}>&divide;</button>
```
Comment (Shisui6) [inline src/components/calculator.js]: [Optional] This is one of the project requirements but was missed by the previous reviewer so I highly encourage you to implement it.

- According to the project requirements, your calculator component should have at least 1 child component. Please try to create and import an additional component for use in your calculator component.
image

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @jash-debug, 

# Requires Changes 🍀
Great work on making the changes requested by a previous reviewer 👍
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet🪲.


 ## Highlights
 - Descriptive PR title and summary ✅
 - Follows Github flow ✅
 - Good commit messages ✅

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

_Check the comments under the review._

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Buttons.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import PropTypes from 'prop-types';
+
```
Comment (TadesseDev) [inline src/components/Buttons.js]: - [ ] Good job making the changes @jash-debug, While you did a great job, please Consider removing disabled liters. Remember linter is here to force you to write clean code and help you identify what could possibly be improved on the current codebase.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @jash-debug 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏
 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — APPROVED** [review]
Hi @jash-debug 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Linters are passing 👍 
The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**